File: Wrapping/WrapITK/Java/Tests/ThresholdImageFilter.java
/**
 *  Example on the use of the ThresholdImageFilter
 *
 */

import InsightToolkit.*;

public class ThresholdImageFilter
{
  public static void main( String argv[] )
  {
    System.out.println("ThresholdImageFilter Example");

    itkImageFileReaderIUS2_Pointer reader = itkImageFileReaderIUS2.itkImageFileReaderIUS2_New();
    itkImageFileWriterIUS2_Pointer writer = itkImageFileWriterIUS2.itkImageFileWriterIUS2_New();

    itkThresholdImageFilterIUS2_Pointer filter = itkThresholdImageFilterIUS2.itkThresholdImageFilterIUS2_New();

    filter.SetInput( reader.GetOutput() );
    writer.SetInput( filter.GetOutput() );

    reader.SetFileName( argv[0] );
    writer.SetFileName( argv[1] );

    filter.SetOutsideValue( Integer.parseInt( argv[2] )  );
    filter.ThresholdAbove(  Integer.parseInt( argv[3] )  );

    writer.Update();
  }

}

File: Wrapping/WrapITK/Python/Tests/ThresholdImageFilter.py
#
#  Example on the use of the ThresholdImageFilter
#

import itk
from sys import argv
itk.auto_progress(2)

dim = 2
IType = itk.Image[itk.US, dim]
OIType = itk.Image[itk.UC, dim]

reader = itk.ImageFileReader[IType].New( FileName=argv[1] )
filter  = itk.ThresholdImageFilter[IType].New( reader,
                OutsideValue=eval( argv[3] ) )
# this method can't be called in the New() method because it doesn't
# use the Set notation
filter.ThresholdAbove(  eval( argv[4] )  )

cast = itk.CastImageFilter[IType, OIType].New(filter)
writer = itk.ImageFileWriter[OIType].New( cast, FileName=argv[2] )

writer.Update()



